### PR TITLE
Fix Conda CI dependency resolution and install timeout

### PR DESCRIPTION
**Pull request by glenn-jocher**
The scheduled [Conda job](https://github.com/ultralytics/ultralytics/actions/runs/33954424824/job/101275017118) timed out after two hours resolving packages across `pytorch`, `conda-forge`, and `defaults`, with libmamba errors while explaining an unsatisfiable solve. Even the preceding successful run spent about 23 minutes installing.

Use only conda-forge with strict channel priority and remove inherited defaults during environment setup. Keep the same packages and tests, make installation noninteractive, and limit that step to 20 minutes so solver hangs fail before exhausting the whole job. This follows [conda-forge channel guidance](https://conda-forge.org/docs/user/transitioning_from_defaults/).

Validation on `e94a715a8bbe17b5535aa469b0dbafe78a6bd444`:

- [Full Conda job passed](https://github.com/ultralytics/ultralytics/actions/runs/33963646144/job/101299788921) on Ubuntu/Python 3.12: installation completed in 94 seconds; benchmark and Python train/val/predict/export smoke tests passed; pytest reported 165 passed and 28 skipped.
- Runner logs confirm inherited defaults were removed, strict priority was applied, and PyTorch 2.13.0 / torchvision 0.28.0 CPU builds came from conda-forge.
- Workflow validation and `git diff --check` passed; comparison against main found no new actionlint diagnostics (21 existing shellcheck/custom runner-label warnings).
- Automated review finding addressed and resolved; formatter passed. Final full-diff review: LGTM, no findings.

All PR checks and the dedicated Conda workflow passed on the reviewed head before merge. Final cold full-diff review: LGTM, no findings; zero unresolved review threads.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Updated the scheduled Conda CI environment to use only `conda-forge` with strict channel priority, while making dependency installation noninteractive and time-limited to prevent long solver hangs.

### 📊 Key Changes
- Removed `defaults` and enabled `conda-remove-defaults: true` in `setup-miniconda`.
- Changed channel priority from flexible to `strict` for `conda-forge`.
- Removed the explicit `pytorch` channel from installation while retaining `pytorch-cpu`, `torchvision`, `ultralytics`, and `openvino`.
- Added `--yes` for noninteractive installation and a 20-minute timeout for the Conda dependency step.

### 🎯 Purpose & Impact
- The scheduled Conda job now resolves dependencies exclusively from `conda-forge` and fails earlier if installation hangs.
- The validated Conda job completed installation in 94 seconds and passed its benchmark, smoke tests, and pytest checks.
- No user-facing change; this updates CI environment setup and dependency installation behavior only.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This automated message confirms your contribution was received, and an Ultralytics engineer will assist with the review. To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -611,11 +611,12 @@ jobs:
       - uses: conda-incubator/setup-miniconda@v4
         with:
           python-version: ${{ matrix.python }}
-          channels: conda-forge,defaults
-          channel-priority: true
+          channels: conda-forge
```
Comment (UltralyticsAssistant) [inline .github/workflows/ci.yml:614]: 💡 **MEDIUM**: `setup-miniconda` creates and activates `anaconda-client-env` before the later `conda install` runs. Setting `channels: conda-forge` does not remove an inherited `defaults` entry, so this setup solve can still query or install from `defaults`; `--override-channels` only protects line 619. Add `conda-remove-defaults: "true"` here so the entire environment setup is actually conda-forge-only. ([github.com](https://github.com/conda-incubator/setup-miniconda))

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Reviewed the complete `.github/workflows/ci.yml` diff. The install step is correctly noninteractive, explicitly restricted to conda-forge, and time-limited, but the setup action can still inherit `defaults` during environment creation. One medium-severity correction is needed before LGTM.

<details><summary>💬 Posted 1 inline comment</summary>

- 💡 **MEDIUM** `.github/workflows/ci.yml:614` `setup-miniconda` creates and activates `anaconda-client-env` before the later `conda install` runs. Setting `channels: conda-forge` does not remove an inherited `defaults` entry, so this setup solve can still query or install from `defaults`; `--override-channels` only protects line 619. Add `conda-remove-defaults: "true"` here so the entire environment setup is actually conda-forge-only. ([github.com](https://github.com/conda-incubator/setup-miniconda))
</details>

**Comment by UltralyticsAssistant** [comment]
🎉 [PR #26064](https://github.com/ultralytics/ultralytics/pull/26064) has been merged! Thank you, @glenn-jocher, for improving the scheduled Conda CI environment with strict `conda-forge` channel usage, noninteractive installation, and a 20-minute timeout.

> “Great things are done by a series of small things brought together.” — Vincent van Gogh

These focused changes helped the validated Conda job complete installation in 94 seconds while passing its benchmark, smoke tests, and pytest checks.
